src/naive/internals.rs

```diff
@@ -271,27 +270,30 @@ pub(super) struct Of(pub(crate) u32);
 
 impl Of {
     #[inline]
-    pub(super) fn new(ordinal: u32, YearFlags(flags): YearFlags) -> Option<Of> {
+    pub(super) const fn new(ordinal: u32, YearFlags(flags): YearFlags) -> Option<Of> {
         match ordinal <= 366 {
-            true => Some(Of((ordinal << 4) | u32::from(flags))),
+            true => Some(Of((ordinal << 4) | flags as u32)),
             false => None,
         }
     }
 
     #[inline]
-    pub(super) fn from_mdf(Mdf(mdf): Mdf) -> Of {
+    pub(super) const fn from_mdf(Mdf(mdf): Mdf) -> Of {
         let mdl = mdf >> 3;
-        match MDL_TO_OL.get(mdl as usize) {
-            Some(&v) => Of(mdf.wrapping_sub((i32::from(v) as u32 & 0x3ff) << 3)),
-            None => Of(0),
```

Comment:
These three lines where using `array.get()` and returning a safe value instead of panicking on out-of-bounds. Yet the index value is created in this module, and always in bounds.

 I suspect the reason here is not just to avoid a potential panic, but to avoid the creation of landings pads if the compiler can't elide the bounds check.

In my opinion it would be fine to simplify this code to direct array indexing, but I kept the functionality. Note that `array.get()` is unavailable in const, so this needs a manual bounds check.

Reply:
Ai. I was wrong, this code is just ancient. In various places it is returning `0` instead of `Option`, and you are supposed to call `valid()` afterwards 😞. Let me see if I can clean that up a little.

Reply:
we may even have some correctness gains by separating these values out to separate modules, with the modules only exposing APIs that allow valid values (in cases where a const API still requires things like .valid())